src/algorithm/area.rs

```diff
@@ -25,8 +25,8 @@ fn get_linestring_area<T>(linestring: &LineString<T>) -> T where T: Float {
         return T::zero();
     }
     let mut tmp = T::zero();
-    for (p1, p2) in linestring.0.iter().zip(linestring.0[1..].iter()) {
-        tmp = tmp + (p1.x() * p2.y() - p2.x() * p1.y());
+    for ps in linestring.0.windows(2).into_iter() {
```

Comment:
FYI, you should be able to just do `for ps in linestring.0.windows(2)` here. Using a `for` loop will automatically call [`IntoIterator`](https://doc.rust-lang.org/std/iter/trait.IntoIterator.html) for the item being interated over

https://doc.rust-lang.org/std/iter/index.html#for-loops-and-intoiterator